src/go/k8s/apis/redpanda/v1alpha1/console_enterprise_types.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import corev1 "k8s.io/api/core/v1"
+
+// Enterprise defines configurable fields for features that require license
+type Enterprise struct {
+	// Console uses role-based access control (RBAC) to restrict system access to authorized users
+	RBAC EnterpriseRBAC `json:"rbac"`
+}
+
+// EnterpriseRBAC defines configurable fields for specifying RBAC Authorization
+type EnterpriseRBAC struct {
+	Enabled bool `json:"enabled"`
+
+	// RoleBindingsRef is the ConfigMap that contains the RBAC file
+	// The ConfigMap should contain "rbac.yaml" key
+	RoleBindingsRef corev1.LocalObjectReference `json:"roleBindingsRef"`
+}
+
+// EnterpriseLogin defines configurable fields to enable SSO Authentication for supported login providers
+type EnterpriseLogin struct {
+	Enabled bool `json:"enabled"`
+
+	// JWTSecret is the Secret that is used to sign and encrypt the JSON Web tokens that are used by the backend for session management
+	// If not provided, the default key is "jwt"
+	JWTSecretRef SecretKeyRef `json:"jwtSecretRef"`
+
+	Google *EnterpriseLoginGoogle `json:"google,omitempty"`
+}
+
+// EnterpriseLoginGoogle defines configurable fields for Google provider
+type EnterpriseLoginGoogle struct {
+	Enabled bool `json:"enabled"`
```

Comment:
We add this in API spec instead of inferring from the code so user can disable this functionality without removing the other fields:
```
...
login:
  enabled: false
  jwtSecretRef:
    name: "test-console-jwt"
    namespace: "default"
  google:
    enabled: true
    clientCredentialsRef:
      name: "test-console-google"
      namespace: "default"
    directory:
      serviceAccountRef:
        name: test-console-google-sa
      targetPrincipal: alex@redpanda.com
```

Reply:
I see your point. 

@alenkacz @mo-redpanda @joejulian @nicolaferraro 

From user perspective which option would be better from your point of view?

- dedicated enable field - other filed can be defined and if user misses `enable: true` then google SSO will not work
- without enable field - if user defines google sso configuration enabled field in console configuration will be assumed

Reply:
I had the same discussion with Moritz when I added the config in Console. Both have pros and cons, but in the end we decided to go with the more verbose/explicit version that uses the bool for several blocks that are otherwise optional. That was helpful for e.g. `tls`, where you can enable/enforce TLS even if you don't require further input like publicly signed TLS.

This basically allows us to have very strict config validation. I think it makes sense if the operator mimics the Console config and doesn't deviate from the Console config as this could cause issues/challenges at some places.

Reply:
Regarding:

> dedicated enable field - other filed can be defined and if user misses enable: true then google SSO will not work

If no SSO provider is enabled Console wouldn't start as it validates whether it has at least one SSO provider that is enabled.